File: packages/javascript/lib/array/arrMove.ts
import type { PositiveNumber } from '../_types/global.js';

export function arrMove<N extends number>(
	array: any[],
	from: PositiveNumber<N>,
	to: PositiveNumber<N>
): any[] {
	const arrayLength = array.length;

	if (arrayLength <= from || arrayLength <= to) {
		throw new Error('Invalid move params');
	}

	// Move within a copy so the caller's array is left untouched.
	const newArray = [...array];

	newArray.splice(to, 0, newArray.splice(from, 1)[0]);

	return newArray;
}

File: packages/python/qsu/array/arrMove.py
def arrMove(array: list, fromIndex: int, to: int) -> list:
	arrayLength = len(array)

	if arrayLength <= fromIndex or arrayLength <= to:
		raise Exception('Invalid move params')

	# Move within a copy so the caller's list is left untouched.
	newArray = list(array)

	newArray.insert(to, newArray.pop(fromIndex))

	return newArray
